azext_iot/central/commands_device.py

```diff
@@ -5,31 +5,69 @@
 # --------------------------------------------------------------------------------------------
 # Dev note - think of this as a controller
 
+from azext_iot.central.common import EDGE_ONLY
 from azext_iot.central.models.devicetwin import DeviceTwin
-from azext_iot.central.providers import CentralDeviceProvider
-from azure.cli.core.azclierror import InvalidArgumentValueError, RequiredArgumentMissingError
-from typing import Union, List
+from azext_iot.central.models.edge import EdgeModule
+from azext_iot.central.providers import (
+    CentralDeviceProvider,
+    CentralDeviceTemplateProvider,
+)
+
+from typing import Union, List, Any
+from azure.cli.core.azclierror import (
+    InvalidArgumentValueError,
+    RequiredArgumentMissingError,
+    ResourceNotFoundError,
+    ForbiddenError,
+)
 from azext_iot.central.models.v1 import DeviceV1
 from azext_iot.central.models.preview import DevicePreview
 from azext_iot.central.models.v1_1_preview import DeviceV1_1_preview
 from azext_iot.common import utility
 from azext_iot.constants import CENTRAL_ENDPOINT
 from azext_iot.central.models.enum import ApiVersion
+from knack.log import get_logger
+
+logger = get_logger(__name__)
 
 DeviceType = Union[DevicePreview, DeviceV1, DeviceV1_1_preview]
 
 
 def list_devices(
     cmd,
     app_id: str,
+    edge_only=False,
     token=None,
     central_dns_suffix=CENTRAL_ENDPOINT,
     api_version=ApiVersion.v1.value,
 ) -> List[DeviceType]:
     provider = CentralDeviceProvider(
         cmd=cmd, app_id=app_id, token=token, api_version=api_version
     )
-    return provider.list_devices(central_dns_suffix=central_dns_suffix)
+    devices = provider.list_devices(
+        filter=EDGE_ONLY, central_dns_suffix=central_dns_suffix
```

Comment:
Does this filter mean that the only devices returned will be edge devices always?

Reply:
good catch. bad copy/paste